internal/credential/vault/credential.go

```diff
@@ -10,7 +10,37 @@ import (
 	"google.golang.org/protobuf/proto"
 )
 
-const externalIdSentinel = "\ufffenone"
+// A CredentialStatus represents the status of a vault credential.
+type CredentialStatus string
+
+const (
+	// ActiveCredential represents a vault credential that is being used in
+	// an active session. Credentials in this state are renewed before they
+	// expire.
+	ActiveCredential CredentialStatus = "active"
+
+	// RevokeCredential represents a vault credential that needs to be
+	// revoked.
+	RevokeCredential CredentialStatus = "revoke"
+
+	// RevokedCredential represents a credential that has been revoked. This is a
+	// terminal status. It does not transition to ExpiredCredential.
+	RevokedCredential CredentialStatus = "revoked"
+
+	// ExpiredCredential represents a credential that expired. This is a terminal
+	// status. It does not transition to RevokedCredential.
+	ExpiredCredential CredentialStatus = "expired"
+)
+
+const (
+	// TODO(mgaffney) 06/2021: credentials with an externalIdSentinel value
```

Comment:
Not following this TODO, it seems you are handling this below

Reply:
Fixed. Ty!